tests/unit/s2n_async_pkey_test.c

```diff
@@ -300,6 +424,66 @@ int main(int argc, char **argv)
 
     EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
 
+    struct s2n_hash_state digest = { 0 };
+    EXPECT_SUCCESS(s2n_hash_new(&digest));
+    EXPECT_SUCCESS(s2n_hash_init(&digest, S2N_HASH_SHA1));
+    EXPECT_SUCCESS(s2n_hash_update(&digest, test_digest_data, test_digest_size));
+
+    /* Test: signature offload. */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+        conn->config->async_pkey_cb = async_pkey_signature_callback;
+
+        EXPECT_FALSE(s2n_result_is_ok(s2n_async_pkey_sign(conn, S2N_SIGNATURE_ECDSA, &digest, s2n_async_sign_complete)));
+        EXPECT_TRUE(s2n_errno == S2N_ERR_ASYNC_BLOCKED);
+
+        EXPECT_SUCCESS(s2n_async_pkey_op_apply(pkey_op, conn));
+        EXPECT_SUCCESS(s2n_async_pkey_op_free(pkey_op));
+
+        EXPECT_SUCCESS(s2n_connection_free(conn));
+    }
+
+    /* Test: decrypt offload. */
+    {
+        struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
+        EXPECT_NOT_NULL(conn);
+        conn->config->async_pkey_cb = async_pkey_decrypt_callback;
+
+        struct s2n_blob encrypted_data = { 0 };
+        EXPECT_SUCCESS(s2n_blob_init(&encrypted_data, test_encrypted_data, test_encrypted_size));
+
+        struct s2n_blob decrypted_data = { 0 };
+        /* Re-use the encrypted data buffer to make sure that the data was actually transformed in the callback. 
+         * If we filled this with the decrypted data, we would not if the decryption happened in the callback. */
```

Comment:
You're missing a word here. Is it "know"? I'm not sure I understand this comment: how does reusing the data verify anything? Wouldn't that make it less clear?
```suggestion
        /* Re-use the encrypted data buffer to make sure that the data was actually transformed in the callback. 
         * If we filled this with the decrypted data, we would not know if the decryption happened in the callback. */
```

Reply:
Yes sorry it is a bit confusing. I am re-using this data to initialize the blob, as the API expect's a malloc'd blob. Really it could be any garbage data, which may be clearer